string.c

```diff
@@ -3073,20 +3073,20 @@ rb_str_set_len(VALUE str, long len)
 VALUE
 rb_str_resize(VALUE str, long len)
 {
-    long slen;
-    int independent;
-
     if (len < 0) {
         rb_raise(rb_eArgError, "negative string size (or size too big)");
     }
 
-    independent = str_independent(str);
-    ENC_CODERANGE_CLEAR(str);
```

Comment:
`rb_str_resize` is public, but its documentation doesn't say anything about clearing the coderange. So I don't know if this change is OK, or if I should preserve this behavior for C-exts.